lib/net.js

```diff
@@ -1112,7 +1112,26 @@ Server.prototype._listen2 = function(address, port, addressType, backlog, fd) {
   // In the case of a server sent via IPC, we don't need to do this.
   if (!self._handle) {
     debug('_listen2: create a handle');
-    var rval = createServerHandle(address, port, addressType, fd);
+
+    var rval = null;
+
+    if (!address && !util.isNumber(fd)) {
+      rval = createServerHandle('::', port, 6, fd);
```

Comment:
Oh, you're right.  Okay, complaint withdrawn. :-)

Maybe you can swap the clauses of the `if !(self.handle)` statement.  The consequent is big enough now that it doesn't fit on a single page anymore, somewhat hiding the alternate.